opentelemetry-sdk/src/resource/env.rs

```diff
@@ -76,15 +76,15 @@ pub struct SdkProvidedResourceDetector;
 impl ResourceDetector for SdkProvidedResourceDetector {
     fn detect(&self, _timeout: Duration) -> Resource {
         Resource::new(vec![KeyValue::new(
-            "service.name",
+            super::SERVICE_NAME,
             env::var(OTEL_SERVICE_NAME)
                 .ok()
                 .filter(|s| !s.is_empty())
                 .map(Value::from)
                 .or_else(|| {
                     EnvResourceDetector::new()
                         .detect(Duration::from_secs(0))
-                        .get(Key::new("service.name"))
+                        .get(Key::new(super::SERVICE_NAME))
```

Comment:
this is the name of the env variable, right? Not semantic convention?

Reply:
Not the env variable, but the service name key value within the `OTEL_RESOURCE_ATTRIBUTES` key value.

This is the fallback value retrieved from the EnvResourceDetector, in which the service name can be defined directy on the key values list: `OTEL_RESOURCE_ATTRIBUTES="service.name=service1,another=test"`

In this case we try to get the "service.name" (semantic convention) from the parsed key values.